nat-lab/tests/utils/analytics/event_validator.py

```diff
@@ -267,52 +279,238 @@ def validate(self, event):
         return self._validator.validate(event.heartbeat_interval)
 
 
-class ReceivedDataValidator:
-    def __init__(self, exists=True):
-        self._validator = StringValidator(exists=exists)
+class RttValidator:
+    def __init__(
+        self,
+        exists=True,
+        members: Optional[List[str]] = None,
+        equals="",
+        contains: Optional[List[str]] = None,
+        does_not_contain: Optional[List[str]] = None,
+    ):
+        self._members = members
+        if not members:
+            self._validator = StringValidator(exists=exists)
+        else:
+            self._validator = StringValidator(
+                exists=exists,
+                equals=equals,
+                contains=contains,
+                does_not_contain=does_not_contain,
+            )
 
     def validate(self, event):
-        return self._validator.validate(event.received_data)
+        if not self._members:
+            return self._validator.validate(event.rtt)
 
+        rtt_list = event.rtt.split(",")
+        peers = event.members.split(",")
```

Comment:
Am I missing something, why is this `.members` not something like `contex_application_libtelioapp_config_external_links`

Reply:
Ok so it mapped in Event object, but here you are processing internal members. shouldn't it be `external_links`

Reply:
Ok so proccess extranl members does this already

Reply:
I think in genral it would be good to add typing hints in validator, and move that process external peers into some computed property on event, like all_nodes, all_rrt, all_x.

But we can queue this for future.